src/compiler/checker.ts

```diff
@@ -3007,21 +3054,32 @@ namespace ts {
                 // fast path, `symbol` is either already the alias or isn't aliased
                 return symbol;
             }
+            // Check if container is a thing with an `export=` which points directly at `symbol`, and if so, return
+            // the container itself as the alias for the symbol
+            const exportEquals = container.exports && container.exports.get(InternalSymbolName.ExportEquals);
+            if (exportEquals && getSymbolIfSameReference(exportEquals, symbol)) {
+                return container;
+            }
             const exports = getExportsOfSymbol(container);
             const quick = exports.get(symbol.escapedName);
-            if (quick && symbolRefersToTarget(quick)) {
+            if (quick && getSymbolIfSameReference(quick, symbol)) {
                 return quick;
             }
             return forEachEntry(exports, exported => {
-                if (symbolRefersToTarget(exported)) {
+                if (getSymbolIfSameReference(exported, symbol)) {
                     return exported;
                 }
             });
+        }
 
-            function symbolRefersToTarget(s: Symbol) {
-                if (s === symbol || resolveSymbol(s) === symbol || resolveSymbol(s) === resolveSymbol(symbol)) {
-                    return s;
-                }
+        /**
+         * Checks if two symbols, through aliasing and/or merging, refer to the same thing
+         */
+        function getSymbolIfSameReference(s1: Symbol, s2: Symbol) {
+            if (getMergedSymbol(s1) === getMergedSymbol(s2) ||
+                getMergedSymbol(resolveSymbol(s1)) === getMergedSymbol(s2) ||

```

Comment:
let *m*=`getMergedSymbol` and *r*=`resolveSymbol` in ...
why have `m(r(s1)) === m(s2)`  if `r(s2) === s2` for non-aliases? The next line should suffice, right?

Reply:
In theory, `getMergedSymbol(resolveSymbol(getMergedSymbol(s1))) === getMergedSymbol(resolveSymbol(getMergedSymbol(s2)))` handles everything. IIRC when I first wrote it I was trying to avoid calls if possible? IDK, there's no real reason to.